### PR TITLE
Extract TradeItSDK class that holds launcher statically

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItSDK.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+@objc public class TradeItSDK: NSObject {
+    private static var apiKey: String?
+    private static var environment: TradeitEmsEnvironments?
+
+    public static let launcher = TradeItLauncher()
+    private (set) public static var linkedBrokerManager: TradeItLinkedBrokerManager!
+    private (set) public static var marketDataService: TradeItMarketService!
+    private (set) public static var brokerCenterService: TradeItBrokerCenterService!
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: @mitochondrion any ideas for how to warn if they haven't called `configure` before? Computed properties won't allow you to throw an exception or anything like that.

File: TradeItIosTicketSDK2/TradeItSDK.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
+@objc public class TradeItSDK: NSObject {
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Can you actually check this branch out and run it. It compiles and runs, but I still get compilation ERRORS about `TradeItSDK` not found in various classes. It's really weird. I cleaned the build completely but I still see the errors in xcode.

File: TradeItIosTicketSDK2/TradeItSDK.swift (hunk 1 above)
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Live example: http://swiftlang.ng.bluemix.net/#/repl/584b448223d4e234e48b7eae

```swift
class Foo
    private static var _thing: Int?
    public static var thing: Int {
        get {
            precondition(_thing != nil, "ERROR: Foo.thing referenced before calling Foo.config()!")
            return _thing!
        }
    }

    public static func config() {
        Foo._thing
    }
}
```

File: TradeItIosTicketSDK2/TradeItSelectBrokerViewController.swift [hunk 3]
```diff
@@ -5,7 +5,7 @@ class TradeItSelectBrokerViewController: TradeItViewController, UITableViewDeleg
     @IBOutlet weak var brokerTable: UITableView!
     var delegate: TradeItSelectBrokerViewControllerDelegate?
     var alertManager = TradeItAlertManager()
-    var linkedBrokerManager: TradeItLinkedBrokerManager = TradeItLauncher.linkedBrokerManager
+    var linkedBrokerManager: TradeItLinkedBrokerManager = TradeItSDK.linkedBrokerManager
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSelectBrokerViewController.swift]: Yeah that's fine - I thought you preferred this way so was sticking with the pattern

File: TradeItIosTicketSDK2/TradeItSDK.swift (hunk 1 above)
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Hmm, interesting. I can't think of a better way of getting the desired effect. Let's do it.

File: TradeItIosTicketSDK2/TradeItSDK.swift (hunk 2 above)
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: @mitochondrion pls do this. I know it makes no sense. Literally just check it out and run it and see if you get any compiler errors.

File: TradeItIosTicketSDK2/TradeItSDK.swift [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    private static var apiKey: String?
+    private static var environment: TradeitEmsEnvironments?
+
+    public static let launcher = TradeItLauncher()
+    private (set) public static var linkedBrokerManager: TradeItLinkedBrokerManager!
+    private (set) public static var marketDataService: TradeItMarketService!
+    private (set) public static var brokerCenterService: TradeItBrokerCenterService!
+
+    public static func configure(apiKey: String, environment: TradeitEmsEnvironments = TradeItEmsProductionEnv) {
+        if self.apiKey == nil {
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Sure. It's a little more obvious but also adds a variable - I think it's ok either way.

File: TradeItIosTicketSDK2/TradeItSDK.swift (hunk 2 above)
Comment (mitochondrion) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Nope.  No errors.

**Pull request merged**